kolena/_experimental/dataset/_dataset.py

```diff
@@ -113,15 +121,52 @@ def _to_deserialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     return result
 
 
-def register_dataset(name: str, df: pd.DataFrame) -> None:
+def _upload_dataset_chunk(df: pd.DataFrame, load_uuid: str, id_fields: List[str]) -> None:
+    df_serialized_datapoint = _to_serialized_dataframe(df, column=COL_DATAPOINT)
+    df_serialized_datapoint_id_object = _to_serialized_dataframe(df[sorted(id_fields)], column=COL_DATAPOINT_ID_OBJECT)
+    df_serialized = pd.concat([df_serialized_datapoint, df_serialized_datapoint_id_object], axis=1)
+
+    upload_data_frame(df=df_serialized, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
+
+
+def load_dataset(name: str) -> Optional[EntityData]:
+    response = krequests.put(Path.LOAD_DATASET, json=asdict(LoadDatasetByNameRequest(name=name)))
+    if response.status_code == requests.codes.not_found:
+        return None
+
+    response.raise_for_status()
+
+    return from_dict(EntityData, response.json())
+
+
+def register_dataset(name: str, df: Union[Iterator[pd.DataFrame], pd.DataFrame], id_fields: List[str]) -> None:
     """
-    Create or update a dataset with datapoints.
+    Create or update a dataset with datapoints and id_fields. If the dataset already exists, in order to associate the
+    existing result with the new datapoints, the id_fields need be the same as the existing dataset.
+
+    :param name: name of the dataset
+    :param df: an iterator of pandas dataframe or a pandas dataframe, you can pass in the iterator if you want to have
+                batch processing,
+                 example iterator usage: csv_reader = pd.read_csv("PathToDataset.csv", chunksize=10)
+    :param id_fields: a list of id fields, this will be used to link the result with the datapoints
+    :return None
     """
     load_uuid = init_upload().uuid
-
-    df_serialized = _to_serialized_dataframe(df, column=COL_DATAPOINT)
-    upload_data_frame(df=df_serialized, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
-    request = RegisterRequest(name=name, uuid=load_uuid)
+    existing_dataset = load_dataset(name)
+    existing_id_fields = []
+    if existing_dataset:
```

Comment:
Any reason to continue if dataset does not exist?

Reply:
This function allows both modify and creation, `register_dataset` is a misleading name, but changing it would also be breaking. Can we change to `create_or_update_dataset`?

Reply:
temporary insanity, mixed up with `test`. Sorry for the confusion. No issue here.